framework/src/st_things/things_stack/cloud/cloud_connector.c

```diff
@@ -83,42 +80,30 @@ int CICheckDomain(const char *DomainName, char **pIP)
 		}
 	} 
 
-	THINGS_LOG_E(TAG, " Failed to get the IP, hard coding the ip");
-	snprintf(ipbuffer, sizeof(ipbuffer), "%s", "13.124.51.231");
-
-	usleep(500);
-	if (pIP != NULL) {
-		*pIP = things_strdup(ipbuffer);
-	}
+	THINGS_LOG_E(TAG, " Failed to get the IP");
 
-	return 0;
+	return -1;
 }
 
 /**
  * @brief              Entry Point for checking about internet & Cloud connection.
  * @param DomainName   wanted Domain-Name string to look-up IP address. (unit: const char*)
  * @param pIP          Variable pointer to store Founded IP address. (unit: char**)
  * @return             success : return 0 \n
- *                     failure : return > 0
+ *                     failure : return -1
  */
 int ci_connection_pre_check(const char *DomainName, char **pIP)
 {
-	int n_err = 0;				// Success return value
-
 	THINGS_LOG_D(TAG, "DomainName=%s, pIP=0x%X", DomainName, pIP);
-
-	if (CICheckDomain("samsung.com", NULL) != 0) {
-		if (DomainName == NULL) {
-			THINGS_LOG_E(TAG, "AP is Not Connected to Internet.");
-			return -1;
-		}
-	}
-
-	if (DomainName != NULL) {
+	int n_err = 0;
+	n_err = CICheckDomain("samsung.com", NULL);
+	if (n_err == -1) {
+		return -1;
+	} else if (n_err == 0 && DomainName != NULL) {
 		n_err = CICheckDomain(DomainName, pIP);
 		THINGS_LOG_D(TAG, "Found IP = %s, n_err = %d", *pIP, n_err);
 	}
-
+	
```

Comment:
please , remove the white space

Reply:
Thanks.

Reply:
To understand code more easily, how about below?
```
int n_err = -1;
n_err = CICheckDomain("samsung.com", NULL);
if (n_err != 0) {
    THINGS_LOG_E(TAG, "AP is Not Connected to Internet");
    return n_err;
}
/** If AP Connected but DomainName is Null, do not check Domain return Success **/
if (DomainName == NULL) {
    THINGS_LOG_V(TAG, "AP Connected but DomainName is Null\n");
     n_err = 0;
} else {
    n_err = CICheckDomain(DomainName, pIP);
    THINGS_LOG_D(TAG, "Found IP = %s, n_err = %d", *pIP, n_err);
}
return n_err;
```

Reply:
good.
This is very clear.